qwc-front.php

```diff
@@ -60,6 +60,27 @@ function qwc_add_filters_front() {
 	//foreach ( $url_filters as $name => $priority ) {
 	//	add_filter( $name, 'qtranxf_convertURL', $priority );
 	//}
+
+	/* Fix the product categories and tags (displayed above the "additional informations" tab) */
+	add_filter( 'wp_get_object_terms', function ( $terms ) {
+		foreach ( $terms as $term ) {
+			if ( $term->taxonomy == 'product_cat' || $term->taxonomy == 'product_tag' ) {
+				$term->name = qtranxf_useCurrentLanguageIfNotFoundUseDefaultLanguage( $term->name );
+			}
+		}
+
+		return $terms;
+	} );
+
+	/* Fix the product attributes (displayed in the "additional informations" tab) */
+	add_filter( 'woocommerce_attribute', function ( $text ) {
+		$values = explode( ', ', $text );
+		foreach ( $values as $i => $val ) {
+			$values[ $i ] = qtranxf_useCurrentLanguageIfNotFoundUseDefaultLanguage( $val );
+		}
+
+		return implode( ', ', $values );
+	} );
 }
 qwc_add_filters_front();
```

Comment:
That is what ending tag `[:]` was supposed to fix. `qtranxf_useCurrentLanguageIfNotFoundUseDefaultLanguage` supposed to fork on the whole $text properly, if [:] is there. Does not it work? The old fields may need to be re-saved in order for `[:]` to be added at the end of each item. Or is there other problem?